Cargo.toml

```diff
@@ -13,5 +13,24 @@ documentation = "https://jellyfish.docs.espressosys.com"
 repository = "https://github.com/EspressoSystems/jellyfish"
 
 [workspace.dependencies]
-itertools = { version = "0.12.1", default-features = false }
-tagged-base64 = { git = "https://github.com/EspressoSystems/tagged-base64.git", tag = "0.3.4" }
+ark-bls12-377 = { version = "0.4.0", default-features = false, features = [ "curve" ] }
+ark-bls12-381 = { version = "0.4.0", default-features = false, features = [ "curve" ] }
+ark-bn254 = { version = "0.4.0", default-features = false, features = [ "curve" ] }
+ark-bw6-761 = { version = "0.4.0", default-features = false }
+ark-ec = { version = "0.4.0", default-features = false }
+ark-ff = { version = "0.4.0", default-features = false, features = [ "asm" ] }
+ark-poly = { version = "0.4.0", default-features = false }
+ark-serialize = { version = "0.4.0", default-features = false }
+ark-std = { version = "0.4.0", default-features = false }
+derivative = { version = "2", features = ["use_core"] }
+digest = { version = "0.10", default-features = false }
+displaydoc = { version = "0.2.3", default-features = false }
+hashbrown = "0.14.3"
+merlin = { version = "3.0.0", default-features = false }
+num-bigint = { version = "0.4", default-features = false }
+rand_chacha = { version = "0.3.1", default-features = false }
+serde = { version = "1.0", default-features = false, features = [ "derive", "rc" ] }
+sha2 = { version = "0.10", default-features = false }
+sha3 = { version = "0.10", default-features = false }
+itertools = { version = "0.12", default-features = false }
+tagged-base64 = { version = "0.4" }
```

Comment:
we have to be a little conscious about bringing these depenedices up to workspace level since they will be built and compiled even if i only need a specific workspace member crate that doesn't depend on them (thus making the compilation slower)

specifically, I think the curve crates could be pushed down?
(maybe debatable, the only reason our generic code still depend on these curves is because we are implementing specialized [rescue parameters](https://github.com/EspressoSystems/jellyfish/tree/main/primitives/src/rescue/rescue_constants) for them, which IMO should be hidden behind feature flags)

maybe for now, this is not so painful and we could afford importing them all the time. but just keep this caveat in mind if we were to add even more curves support in the future.